src/corelibs/U2View/src/ov_phyltree/TreeViewer.cpp

```diff
@@ -650,28 +668,58 @@ void TreeViewerUI::updateSettings() {
         scene()->update();
     }
 }
-void TreeViewerUI::updateTextSettings() {
-    QList<QGraphicsItem *> updatingItems = scene()->selectedItems();
-    QColor curColor = qvariant_cast<QColor>(getOptionValue(LABEL_COLOR));
-    if (updatingItems.isEmpty()) {
-        updatingItems = items();
-
-        QList<QGraphicsItem *> legendChildItems = legend->childItems();
-        if (!legendChildItems.isEmpty()) {
-            QGraphicsSimpleTextItem *legendText = dynamic_cast<QGraphicsSimpleTextItem *>(legendChildItems.first());
-            if (legendText) {
-                legendText->setBrush(curColor);
-            }
+void TreeViewerUI::getSelectedBranchSettings(QColor &color, QFont &font) const {
+    QList<QGraphicsItem *> selectedItems = scene()->selectedItems();
+    if (selectedItems.isEmpty()) {
+        return;
+    }
+    int childCount = 0;
+    GraphicsBranchItem *rootItem = nullptr;
+    for (QGraphicsItem *graphItem : qAsConst(selectedItems)) {
+        auto *branchItem = dynamic_cast<GraphicsBranchItem *>(graphItem);
+        CHECK(branchItem != nullptr, )
+        if (branchItem->getChildItems().size() >= childCount) {
+            childCount = branchItem->getChildItems().size();
+            rootItem = branchItem;
```

Comment:
Question: why the rootItem is the last branchItem?

Reply:
Actually, there is only one root item in selection with the largest number of children.